test.sh

```diff
@@ -28,29 +28,18 @@ if [ -r /proc/brcm_monitor0 ]; then
   exit 2
 fi
 
-cd $(dirname $0)
+cd $(dirname $0)/v4
 DIR=$(mktemp -d -t softhXXXX)
 
-# Travis doesn't currently offer softhsm2 (it's not in Ubuntu Trusty), so we run
-# softhsm2 if it's available (e.g., locally), and softhsm otherwise (e.g. in
-# Travis). ECDSA is only supported in SoftHSMv2 so we skip the ECDSA benchmark
-# when it's not available.
-if $(type softhsm2-util 2>/dev/null >&2) ; then
-  MODULE=${MODULE:-/usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so}
-  export SOFTHSM2_CONF=${DIR}/softhsm.conf
-  echo directories.tokendir = ${DIR} > ${SOFTHSM2_CONF}
-  softhsm2-util --module ${MODULE} --slot 0 --init-token --label silly_signer --pin 1234 --so-pin 1234
-  softhsm2-util --module ${MODULE} --slot 0 --import v4/testdata/silly_signer.key --label silly_signer_key --pin 1234 --id F00D
-  softhsm2-util --module ${MODULE} --slot 1 --init-token --label entropic_ecdsa --pin 1234 --so-pin 1234
-  softhsm2-util --module ${MODULE} --slot 1 --import v4/testdata/entropic_ecdsa.key --label entropic_ecdsa_key --pin 1234 --id C0FFEE
-else
-  MODULE=${MODULE:-/usr/lib/softhsm/libsofthsm.so}
-  export SOFTHSM_CONF=${DIR}/softhsm.conf
-  SLOT=0
-  echo ${SLOT}:${DIR}/softhsm-slot${SLOT}.db > ${SOFTHSM_CONF}
-  softhsm --module ${MODULE} --slot ${SLOT} --init-token --label silly_signer --pin 1234 --so-pin 1234
-  softhsm --module ${MODULE} --slot ${SLOT} --import v4/testdata/silly_signer.key --label silly_signer_key --pin 1234 --id F00D
-fi
+MODULE=${MODULE:-/usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so}
+export SOFTHSM2_CONF=${DIR}/softhsm.conf
+echo directories.tokendir = ${DIR} > ${SOFTHSM2_CONF}
+softhsm2-util --module ${MODULE} --slot 0 --init-token --label silly_signer --pin 1234 --so-pin 1234 > slot-assignment.txt
+SLOT_ASSIGNMENT=$(sed 's/.*to slot \(.\+\)/\1/' slot-assignment.txt)
+softhsm2-util --module "${MODULE}" --slot "${SLOT_ASSIGNMENT}" --import ../v4/testdata/silly_signer.key --label silly_signer_key --pin 1234 --id F00D
```

Comment:
Why does this `SLOT_ASSIGNMENT` parsing need to happen, instead of just pass `--slot 0` or `--slot 1` to both the --init-token and the --import? Has softhsm2 updated such that it ignores the `--slot` flag to --init-token?

Reply:
Yes, unfortunately. We do the same workaround in Boulder's test/cert-ceremonies/generate.go: https://github.com/letsencrypt/boulder/blob/c931cf06e63df291982e2b72f0d1c4e6368310da/test/cert-ceremonies/generate.go#L20-L30

I'll add to the PR desc.

Reply:
In the ceremony tool, we don't appear to be passing the initial `--slot 0` flag to the `--init-token` command. We're sure that it still randomly generates the slot number even when `--slot 0` is passed?

Reply:
Yes, it emits:

```
$ softhsm2-util --module ${MODULE} --slot 1 --init-token --label silly_signer --pin 1234 --so-pin 1234 
The token has been initialized and is reassigned to slot 1306291535
```

In cert-ceremonies/generate.go, it uses the `--free` flag to get a free slot, which more clearly expresses the intent:

```
$ softhsm2-util --module ${MODULE} --free --init-token --label silly_signer --pin 1234 --so-pin 1234 
Slot 2 has a free/uninitialized token.
The token has been initialized and is reassigned to slot 1414983274
```

I'll do that here too.